site/docs/hosting/firebase.md

```diff
@@ -0,0 +1,171 @@
+8. open `./functions/package.json` and look for the key: `"engines": {"node": "16"}`.
+   The node version should match your installed version of node.
+   Otherwise the project might not run.
+
+## Preparing Your Code
+
+You can use this short example bot as a starting point:
+
+```ts
+import * as functions from "firebase-functions";
+import {webhookCallback, Bot} from "grammy";
+
+const bot = new Bot(<BOT_TOKEN>);
+
+bot.command("start", (ctx) => ctx.reply("Welcome! Up and running."));
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+//you can trigger your function on https://localhost/<firebase-projectname>/us-central1/helloworld during development 
+export const helloWorld = functions.https.onRequest(webhookCallback(bot));
+```
+
+## Development
+
+During development you can use the firebase emulator suite to run your code locally, which is way faster than deploying every change to firebase.
+To install the emulators run:
+
+```shell
+firebase init emulators
+```
+
+The functions emulator should be select already, if not navigate to it using the arrow keys, and select it using `space`.
+You can just hit `enter` for the questions about which port to use for which emulator.
+
+To start the emulators and run :
+
+```shell
+npm run serve
+```
+
+> For some reason the standard configuration of the npm script does not start the typescript compiler in watch mode, so if you use typescript, you also have to run
+
+> ```shell
+> npm run build:watch
+> ```
+
+after the emulators started you should find a line in the console output, that looks like this:
+
+```shell
++  functions[us-central1-helloWorld]: http function initialized (http://127.0.0.1:5001/<firebase-projectname>/us-central1/helloWorld).
+```
+
+This is the local url of your cloud function.
+Right now, your function is only available to the localhost (your computer).
+To actually test your bot, you need to expose your function to the internet, so that the Telegram API can send updates to your bot.\
+There are several services like [localtunnel](https://localtunnel.me/) or [ngrok](https://ngrok.com/) that can help you with that.
+Let's go with localtunnel, because it does not shut down after an hour, if you don't give them money like ngrok does.
+First let's install localtunnel:
+
+```shell
+npm i -g localtunnel
+```
+
+After that we can expose the port `5001` to the internet:
+
+```shell
+lt --port 5001
+```
+
+The local tunnel should give you a unique url like `https://modern-heads-sink-80-132-166-120.loca.lt`
+
+All that's left to do is to tell Telegram where to send the updates.
+You can do this by calling setWebhook.
+For example, open a new tab in your browser and visit this URL:
+
+```text:no-line-numbers
+https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=https://modern-heads-sink-80-132-166-120.loca.lt/hotsouceservice/us-central1/deleteImage
+```
+
+Replace <BOT_TOKEN> with your real bot token, and the `.loca.lt` url with you own url you got from localtunnel.
```

Comment:
````suggestion
I am not sure about the URL. Can you confirm if this revision is correct?

```text:no-line-numbers
https://api.telegram.org/bot<BOT_TOKEN>/setWebhook?url=<LT_URL>
```

Replace <BOT_TOKEN> with your real bot token and <LT_URL> with your own URL that you received from localtunnel.
````

Reply:
You have to append the path to your firebase function (/hotsouceservice/us-central1/helloworld) to the localtunnel url, since that only maps to PORT 5001. But i added the wrong function call at the end of the url anyways, so i'll update that.